ProductsService/ProductsService.Infrastructure/Repositories/QueryRepository.cs

```diff
@@ -0,0 +1,89 @@
+﻿using Microsoft.EntityFrameworkCore;
+using ProductsService.Domain.Abstractions.Database;
+using ProductsService.Domain.Abstractions.Specifications;
+using ProductsService.Domain.Entities.Abstractions;
+using ProductsService.Infrastructure.Data;
+using ProductsService.Infrastructure.Specifications;
+using System.Linq.Expressions;
+
+namespace ProductsService.Infrastructure.Repositories;
+
+internal class QueryRepository<T> : IQueryRepository<T> where T : class, IEntity
+{
+    private readonly QueryDbContext _dbContext;
+    private readonly DbSet<T> _entities;
+
+    public QueryRepository(QueryDbContext dbContext)
+    {
+        _dbContext = dbContext;
+        _entities = _dbContext.Set<T>();
+    }
+
+    public async Task<T?> GetByIdAsync(int id, CancellationToken cancellationToken = default)
+    {
+        var entity = await _entities.FirstOrDefaultAsync(e => e.Id == id);
```

Comment:
I updated the repository to work with the new specification, and now it's not possible to use Find here(Because i use _entites.AsQuerable for navigation properties including)